dialog/fileicon.go

```diff
@@ -0,0 +1,120 @@
+package dialog
+
+import (
+	"image/color"
+	"path/filepath"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+const (
+	fileIconSize = 64
+	fileTextSize = 24
```

Comment:
Shouldn't the theme be the only thing to hold hard-coded sizes?

Reply:
OK, this is the only thing I have not resolved in the updated commits.
Is it worth breaking any existing custom themes (as the API would have to be extended) ?
Can we re-incorporate it in a future (also 1.3) theme API that is extensible so it's not an ongoing challenge?

Reply:
1. No, breaking custom themes doesn't sound like a good idea.
2. Yeah, I think so.